ibis/spark/ddl.py

```diff
@@ -0,0 +1,267 @@
+    _is_quoted,
+    fully_qualified_re,
+)
+from ibis.spark.compiler import _type_to_sql_string
+
+_format_aliases = {'TEXTFILE': 'TEXT'}
+
+
+def _sanitize_format(format):
+    if format is None:
+        return
+    format = format.upper()
+    format = _format_aliases.get(format, format)
+    if format not in (
+        'TEXT',
+        'CSV',
+        'JSON',
+        'JDBC',
+        'PARQUET',
+        'ORC',
+        'HIVE',
+        'DELTA',
+        'LIBSVM',
+    ):
+        raise ValueError('Invalid format: {!r}'.format(format))
+
+    return format
+
+
+def format_tblproperties(props):
+    formatted_props = _format_properties(props)
+    return 'TBLPROPERTIES {}'.format(formatted_props)
+
+
+def _format_properties(props):
+    tokens = []
+    for k, v in sorted(props.items()):
+        tokens.append("  '{}'='{}'".format(k, v))
+
+    return '(\n{}\n)'.format(',\n'.join(tokens))
+
+
+class CreateTable(impala_ddl.CreateTable):
+
+    """Create a table"""
+
+    def __init__(
+        self,
+        table_name,
+        database=None,
+        format='parquet',
+        can_exist=False,
+        path=None,
+        tbl_properties=None,
+    ):
+        super().__init__(
+            table_name,
+            database=database,
+            external=False,
+            format=format,
+            can_exist=can_exist,
+            partition=None,
+            path=path,
+            tbl_properties=tbl_properties,
+        )
+
+    def _storage(self):
+        # By the time we're here, we have a valid format
+        return 'USING {}'.format(self.format)
+
+
+class CreateTableWithSchema(CreateTable):
+    def __init__(self, table_name, schema, **kwargs):
+        super().__init__(table_name, **kwargs)
+        self.schema = schema
+
+    @property
+    def _pieces(self):
+        yield format_schema(self.schema)
+        yield self._storage()
+        yield self._location()
+
+
+class CTAS(CreateTable):
+
+    """
+    Create Table As Select
+    """
+
+    def __init__(
+        self,
+        table_name,
+        select,
+        database=None,
+        format='parquet',
+        can_exist=False,
+        path=None,
+    ):
+        super().__init__(
+            table_name,
+            database=database,
+            format=format,
+            can_exist=can_exist,
+            path=path,
+        )
+        self.select = select
+
+    @property
+    def _pieces(self):
+        yield self._storage()
+        yield self._location()
+        yield 'AS'
+        yield self.select.compile()
+
+
+class CreateView(CTAS):
+
+    """Create a view"""
+
+    def __init__(
+        self,
+        table_name,
+        select,
+        database=None,
+        or_replace=True,
+        temporary=False,
+    ):
+        super().__init__(
+            table_name,
+            select,
+            database=database,
+            can_exist=or_replace,
+        )
+        self.temporary = temporary
+
+    @property
+    def _pieces(self):
```

Comment:
This doesn't appear to be necessary.

Reply:
I think it's necessary to prevent `self._storage()` and `self._location()` from getting called by `CTAS._pieces`? Since views don't have any storage format or external location.